$Id$

Make Debian-specific tweaks to xterm:
* Set the default VT widget foreground and background colors to gray90 on
  black, which approximates the color scheme of the Linux virtual console
  (and approximates actual DEC VT terminals far more closely than
  black text on a white background in any case).
* Stop using an Xaw7 gradient for the backgrounds of the xterm menus; it
  produces an unappealing effect if the menus are configured to use a
  larger font than the stock configuration (also, xterm has added items to
  some menus since I last calculated the gradient size and I can't be
  bothered to do it again).
* If the X server is capable of color and has more than 8 planes of color
  depth available, set the menu colors to gray15 on antique white, and
  customize the appearance VT widget's scrollbar.  Otherwise, do not eat up
  precious entries in the color palette.
* Set the "backarrowKeyIsErase" and "ptyInitialErase" parameters to true by
  default.
* Define the "xterm-debian" terminal type in the termcap and terminfo
  files; it is based on "xterm-new" with the exception that the backspace
  is VT220-conformant, as directed by section 9.8 of the Debian Policy
  Manual.

This patch by Branden Robinson.

Index: termcap
===================================================================
--- termcap.orig	2007-03-25 03:34:05.000000000 +0200
+++ termcap	2007-03-25 03:34:37.000000000 +0200
@@ -73,6 +73,14 @@
 	:so=\E[7m:sr=\EM:st=\EH:te=\E[?1049l:ti=\E[?1049h:\
 	:ue=\E[24m:up=\E[A:us=\E[4m:ve=\E[?25h:vi=\E[?25l:
 
+# xterm-debian is identical to xterm-new except that it more strictly
+# conforms to DEC VT220 behavior in the case of the backspace key; this is
+# mandated by Section 9.8 of the Debian Policy Manual.  Also see
+# <http://vt100.net/docs/vt220-rm/>.
+xd|xterm-debian|Debian xterm (VT220-conformant backspace):\
+	:kb=^?:\
+	:tc=xterm-new:
+
 # The xterm-new description has all of the features, but is not completely
 # compatible with vt220.  If you are using a Sun or PC keyboard, set the
 # sunKeyboard resource to true:
@@ -234,5 +242,6 @@
 # is widely used for a variety of incompatible terminal emulations including
 # color_xterm and rxvt.
 v0|xterm|X11 terminal emulator:\
-	:tc=xterm-new:
+	:tc=xterm-debian:
+#	:tc=xterm-new:
 #	:tc=xterm-r6:
Index: terminfo
===================================================================
--- terminfo.orig	2007-03-25 03:34:05.000000000 +0200
+++ terminfo	2007-03-25 03:34:37.000000000 +0200
@@ -823,6 +823,13 @@
 	u8=\E[?1;2c,
 	u9=\E[c,
 	vpa=\E[%i%p1%dd,
+# xterm-debian is identical to xterm-new except that it more strictly
+# conforms to DEC VT220 behavior in the case of the backspace key; this is
+# mandated by the Section 9.8 of the Debian Policy Manual.  Also see
+# <http://vt100.net/docs/vt220-rm/>.
+xterm-debian|Debian xterm (VT220-conformant backspace),
+	kbs=\177,
+	use=xterm-new,
 #
 # The xterm-new description has all of the features, but is not completely
 # compatible with vt220.  If you are using a Sun or PC keyboard, set the
@@ -1602,5 +1609,6 @@
 # is widely used for a variety of incompatible terminal emulations including
 # color_xterm and rxvt.
 xterm|X11 terminal emulator,
-	use=xterm-new,
+	use=xterm-debian,
+#	use=xterm-new,
 #	use=xterm-r6,
Index: XTerm.ad
===================================================================
--- XTerm.ad.orig	2007-03-25 03:34:05.000000000 +0200
+++ XTerm.ad	2007-03-25 03:34:37.000000000 +0200
@@ -116,6 +116,10 @@
 *tek4014*font3: 6x13
 *tek4014*fontSmall: 6x10
 
+! Debian package customizations follow.
+*backarrowKeyIsErase: true
+*ptyInitialErase: true
+
 ! If xterm is built with a toolbar, the widget hierarchy looks like this,
 ! showing widget name / class names.  The complete menu hierarchy is built
 ! at startup because it is needed to make the layout work for the menubar:
Index: XTerm-col.ad
===================================================================
--- XTerm-col.ad.orig	2007-03-25 03:34:05.000000000 +0200
+++ XTerm-col.ad	2007-03-25 03:34:37.000000000 +0200
@@ -7,9 +7,9 @@
 *VT100*boldColors: on
 *VT100*dynamicColors: on
 
-! Uncomment this for "white" text on a dark background.
-!*VT100*foreground: gray90
-!*VT100*background: black
+! Set the default text foreground and background colors.
+*VT100*foreground: gray90
+*VT100*background: black
 
 ! - OR -
 ! Uncomment this for black text on a "white" background.
@@ -114,20 +114,32 @@
 *Form.background: AntiqueWhite
 *form.background: AntiqueWhite
 
+*mainMenu*background: AntiqueWhite
+*mainMenu*foreground: gray15
+*vtMenu*background: AntiqueWhite
+*vtMenu*foreground: gray15
+*fontMenu*background: AntiqueWhite
+*fontMenu*foreground: gray15
+*tekMenu*background: AntiqueWhite
+*tekMenu*foreground: gray15
+
 ! The following two sections take advantage of new features in version 7
-! of the Athena widget library.  Comment them out if you have a shallow
-! color depth.
+! of the Athena widget library; see Xaw(7x).
+
+! This section is commented out because Xaw has no way to dynamically size
+! the gradient.  The dimension parameter may need to be adjusted depending
+! on the font used in the menu.
 !*SimpleMenu*backgroundPixmap:   gradient:vertical?dimension=350&start=gray90&end=gray60
 !*SimpleMenu*foreground:         gray15
 
-!*VT100.scrollbar.thumb:         vlines2
-!*VT100.scrollbar.width:         14
-!*VT100.scrollbar.background:    gray60
-!*VT100.scrollbar.foreground:    rgb:a/5/5
-!*VT100.scrollbar.borderWidth:   0
-!*VT100.scrollbar.displayList:\
-!foreground      gray90;\
-!lines           1,-1,-1,-1,-1,1;\
-!foreground      gray60;\
-!lines           -1,0,0,0,0,-1
+*VT100.scrollbar.thumb:         vlines2
+*VT100.scrollbar.width:         14
+*VT100.scrollbar.background:    gray60
+*VT100.scrollbar.foreground:    rgb:a/5/5
+*VT100.scrollbar.borderWidth:   0
+*VT100.scrollbar.displayList:\
+foreground      gray90;\
+lines           1,-1,-1,-1,-1,1;\
+foreground      gray60;\
+lines           -1,0,0,0,0,-1
 #endif
